### PR TITLE
Add protocol message and protobuf build script

### PR DESCRIPTION
**Pull request by JosephusPaye**
Adds the message from [NUbots/RobocupProtocol](https://github.com/NUbots/RobocupProtocol), with a script to build to JS and TS. Needed for the teams simulator.

### CODE REVIEW

File: src/shared/proto/robocup_extended.proto [hunk 1]
```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
```
Comment (Bidski) [inline src/shared/proto/robocup_extended.proto]: This file shopulent be necessary. It was only done to show that teams can arbitrarily add extensions to the protocol without impacting the main protocol.

**Review by Bidski — CHANGES_REQUESTED** [review]
Everything else looks ok to me. I havent tested it though.

File: src/shared/proto/robocup_extended.proto (hunk 1 above)
Comment (JosephusPaye) [reply src/shared/proto/robocup_extended.proto]: Ah, gotcha

**Review by Bidski — CHANGES_REQUESTED** [review]
`yarn eslint` is not ignoring the generated files
`yarn tscheck` also fails with `Cannot find module '../proto/messages'`
`yarn test` is also failing

**Comment by TrentHouliston** [comment]
Sounds like we need some CI over here
https://github.com/NUbots/dashboard/pull/4

**Review by Bidski — APPROVED** [review]
LGTM

**Pull request merged**